docs-client/src/lib/transports/thrift.ts

```diff
@@ -59,7 +59,7 @@ export default class ThriftTransport extends Transport {
       hdrs.set(name, value);
     }
 
-    return fetch(endpoint.pathMapping, {
+    return fetch(endpointPath || endpoint.pathMapping, {
```

Comment:
I think most logic in `DocService` relies on using the `endpoint`'s path, whereas we are adding a root path unrelated to endpoints.

Instead of reusing `additionalPath` (`urlPath`), what do you think of just passing the `rootPrefix` (`docServiceRoute`) all the way down to the transport layer?
i.e. https://github.com/line/armeria/blob/a86bcc53804e1ecf07aa0c09750a50bf1aeea1be/docs-client/src/lib/transports/annotated-http.ts#L99

This way
- We don't have to worry about edge cases depending on the method type.
- We don't have to worry about url paths that are resolved at the last minute.

Reply:
I did not understand. Do you think it would be better if we pass `rootPrefix` into this method,

https://github.com/line/armeria/blob/4bae026a22ae0cd18d98a42e2cfc135106973e11/docs-client/src/lib/transports/transport.ts#L27-L33

And leave `endpointPath` unmodified? So `doSend` method takes `rootPrefix` instead?

I will push a PR regarding this and LMK if I understood this incorrectly.

Reply:
BTW it seems like this `||` change in thrift should be fine because same logic applies to annotated services?

Reply:
> I will push a PR regarding this and LMK if I understood this incorrectly.

On second thought, perhaps it's better to just add the prefix automatically to the preset UI.
This way if the prefix adding behavior is not what the user wants, worst case the user can just manually remove the prefix.

I'll have to look into a better injection point for this behavior during the weekend, but if you have a better idea which solves this feel free to fix as you'd like. Sorry about the incorrect review 😅

Reply:
No worries, I think your idea makes more sense. Prefix is similar to sending requests to another host (which we were discussing to implement anyway). If we want to show the URL we used to send the request, probably transport should return back the URL used to send because there are additional parsing happening in transport anyway. Should be better to centralize all those URL parsing. I had hard time modifying this code 😄